app/scripts/directives/resourceServiceBindings.js

```diff
@@ -0,0 +1,111 @@
+    project: '<',
+    projectContext: '<',
+    apiObject: '<',
+    createBinding: '&'
+  },
+  templateUrl: 'views/directives/resource-service-bindings.html'
+});
+
+
+function ResourceServiceBindings($filter, DataService, BindingService, CatalogService) {
+  var ctrl = this;
+  ctrl.bindings = [];
+  ctrl.bindableServiceInstances = [];
+  ctrl.serviceClasses = [];
+  ctrl.serviceInstances = [];
+  ctrl.secrets = [];
+  ctrl.showBindings = CatalogService.SERVICE_CATALOG_ENABLED;
+
+  var limitWatches = $filter('isIE')() || $filter('isEdge')();
+  var DEFAULT_POLL_INTERVAL = 60 * 1000; // milliseconds
+  var watches = [];
+  var canI = $filter('canI');
+
+  var refreshSecrets = _.debounce(function(context) {
+    DataService.list("secrets", context, null, { errorNotification: false }).then(function(secretData) {
+      ctrl.secrets = secretData.by("metadata.name");
+    });
+  }, 300);
+
+  var updateBindings = function() {
+    if (!ctrl.apiObject || !ctrl.bindings) {
+      return;
+    }
+
+    // Get only those bindings applicable to the resource
+    ctrl.bindings = BindingService.getBindingsForResource(ctrl.bindings, ctrl.apiObject);
+  };
+
+  var sortServiceInstances = function() {
+    ctrl.bindableServiceInstances = BindingService.filterBindableServiceInstances(ctrl.serviceInstances, ctrl.serviceClasses);
+    ctrl.orderedServiceInstances = BindingService.sortServiceInstances(ctrl.serviceInstances, ctrl.serviceClasses);
+  };
+
+  ctrl.createBinding = function() {
+    ctrl.overlayPanelVisible = true;
+    ctrl.overlayPanelName = 'bindService';
+  };
+
+  ctrl.closeOverlayPanel = function() {
+    ctrl.overlayPanelVisible = false;
+  };
+
+  var updateData = function() {
+    DataService.unwatchAll(watches);
+    watches = [];
+
+    if (CatalogService.SERVICE_CATALOG_ENABLED && canI({resource: 'bindings', group: 'servicecatalog.k8s.io'}, 'watch')) {
+      watches.push(DataService.watch({
+        group: 'servicecatalog.k8s.io',
+        resource: 'bindings'
+      }, ctrl.projectContext, function(bindings) {
+        ctrl.bindings = bindings.by('metadata.name');
+        updateBindings();
+        refreshSecrets(ctrl.projectContext);
+      }, {poll: limitWatches, pollInterval: DEFAULT_POLL_INTERVAL}));
+    }
+
+    // The canI check on watch should be temporary until we have a different solution for handling secret parameters
+    if (CatalogService.SERVICE_CATALOG_ENABLED && canI({resource: 'instances', group: 'servicecatalog.k8s.io'}, 'watch')) {
+      watches.push(DataService.watch({
+        group: 'servicecatalog.k8s.io',
+        resource: 'instances'
+      }, ctrl.projectContext, function(serviceInstances) {
+        ctrl.serviceInstances = serviceInstances.by('metadata.name');
+        sortServiceInstances();
+      }, {poll: limitWatches, pollInterval: DEFAULT_POLL_INTERVAL}));
+    }
+
+    if (CatalogService.SERVICE_CATALOG_ENABLED && canI({resource: 'instances', group: 'servicecatalog.k8s.io'}, 'watch')) {
```

Comment:
Added comment for this, basically no point watching the serviceclasses if we can't watch serviceinstances (removed the extra check and did it all under the same if)